src/Microsoft.AspNetCore.Rewrite/RewriteOptions.cs

```diff
@@ -1,24 +1,49 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Generic;
 using Microsoft.Extensions.FileProviders;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Rewrite.Internal;
 
 namespace Microsoft.AspNetCore.Rewrite
 {
     /// <summary>
     /// Options for the <see cref="RewriteMiddleware"/> 
     /// </summary>
-    public class RewriteOptions
+    public class RewriteOptions : IRewriteOptionsInfrastructure
     {
+        public RewriteOptions()
+        {
+            ((IRewriteOptionsInfrastructure)this).DefaultConfigFileProvider = CreateConfigFileProvider();
```

Comment:
Solves the obfuscates  part, but not the '.Internal' part https://github.com/aspnet/BasicMiddleware/pull/98#discussion_r77885612